From dd57c2b6735d18389e1767a64fa49438bdb5a613 Mon Sep 17 00:00:00 2001
From: Andreas Blaesius <skate4life@gmx.de>
Date: Fri, 9 Jun 2017 14:22:58 +0200
Subject: [PATCH] camera: espresso: revert "add Pixel{X,Y}Dimension and
 Orientation EXIF tags"

- fixes images cropped to square when shooting in portrait

Change-Id: I533ab769f61baebbf7f1eedc9b41c85425aacb9f
(cherry picked from commit c6a06c47329d5e475dc18f6bc138594bf3693b27)
---
 camera/OMXCameraAdapter/OMXExif.cpp | 6 ++++++
 camera/inc/Encoder_libjpeg.h        | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/camera/OMXCameraAdapter/OMXExif.cpp b/camera/OMXCameraAdapter/OMXExif.cpp
index c70c5eb..41935a0 100644
--- a/camera/OMXCameraAdapter/OMXExif.cpp
+++ b/camera/OMXCameraAdapter/OMXExif.cpp
@@ -373,6 +373,7 @@ status_t OMXCameraAdapter::setupEXIF()
              exifTags->eStatusImageHeight = OMX_TI_TagUpdated;
              }
 
+#ifndef CAMERAHAL_PIRANHA
          if ( OMX_TI_TagReadWrite == exifTags->eStatusPixelXDimension &&
               OMX_TI_TagReadWrite == exifTags->eStatusPixelYDimension)
              {
@@ -394,6 +395,7 @@ status_t OMXCameraAdapter::setupEXIF()
              exifTags->usOrientation = 0;
              exifTags->eStatusOrientation = OMX_TI_TagUpdated;
              }
+#endif
 
          if ( ( OMX_TI_TagReadWrite == exifTags->eStatusGpsLatitude ) &&
               ( mEXIFData.mGPSData.mLatValid ) )
@@ -584,18 +586,22 @@ status_t OMXCameraAdapter::setupEXIF_libjpeg(ExifElementsTable* exifTable,
         char temp_value[5];
         snprintf(temp_value, sizeof(temp_value)/sizeof(char), "%lu", (unsigned long)capData->mWidth);
         ret = exifTable->insertElement(TAG_IMAGE_WIDTH, temp_value);
+#ifndef CAMERAHAL_PIRANHA
         if ((NO_ERROR == ret)) {
             ret = exifTable->insertElement(TAG_EXIF_IMAGE_WIDTH, temp_value);
         }
+#endif
      }
 
     if ((NO_ERROR == ret)) {
         char temp_value[5];
         snprintf(temp_value, sizeof(temp_value)/sizeof(char), "%lu", (unsigned long)capData->mHeight);
         ret = exifTable->insertElement(TAG_IMAGE_LENGTH, temp_value);
+#ifndef CAMERAHAL_PIRANHA
         if ((NO_ERROR == ret)) {
             ret = exifTable->insertElement(TAG_EXIF_IMAGE_LENGTH, temp_value);
         }
+#endif
      }
 
     if ((NO_ERROR == ret) && (mEXIFData.mGPSData.mLatValid)) {
diff --git a/camera/inc/Encoder_libjpeg.h b/camera/inc/Encoder_libjpeg.h
index 089b02b..4a7ff39 100644
--- a/camera/inc/Encoder_libjpeg.h
+++ b/camera/inc/Encoder_libjpeg.h
@@ -63,8 +63,10 @@ static const char TAG_FOCALLENGTH[] = "FocalLength";
 static const char TAG_DATETIME[] = "DateTime";
 static const char TAG_IMAGE_WIDTH[] = "ImageWidth";
 static const char TAG_IMAGE_LENGTH[] = "ImageLength";
+#ifndef CAMERAHAL_PIRANHA
 static const char TAG_EXIF_IMAGE_WIDTH[] = "ExifImageWidth";
 static const char TAG_EXIF_IMAGE_LENGTH[] = "ExifImageLength";
+#endif
 static const char TAG_GPS_LAT[] = "GPSLatitude";
 static const char TAG_GPS_LAT_REF[] = "GPSLatitudeRef";
 static const char TAG_GPS_LONG[] = "GPSLongitude";
-- 
2.7.4

